tests/Fixer/Whitespace/HeredocIndentationFixerTest.php

```diff
@@ -176,6 +200,50 @@ public function provideFixCases()
 INPUT
                 ,
             ],
+            [
+                /* EXPECTED */ '
+<?php
+    foo(<<<EOD
+          '.'
+        abc
+          '.'
+        def
+          '.'
+        EOD
+    );',
+                /* INPUT */ '
+<?php
+    foo(<<<EOD
+        '.'
+      abc
+        '.'
+      def
+        '.'
+      EOD
+    );',
+            ],
+            [
+                /* EXPECTED */ '
+<?php
+    foo(<<<EOD
+
```

Comment:
I'm not sure whitespaces that exist before fixing should be removed. IMO the fixer should only avoid adding new ones.

Reply:
I'm also not sure. But what would be the expected result for the given test case for you?
Keep the two spaces in the empty lines? Or change them to 4 spaces, because all other lines also get 2 more spaces indentation?

Reply:
As the fixer is called `HeredocIndentationFixer` it imho just fixes all things wrong with the indentation. Having 2 where it should be 4 is incorrect and should thus be fixed. It then seems weird to pad here and not in fully empty lines, because that would still give an inconsistency: 2 kinds of empty lines, just one with the "prefix" and one without.

I can therefor only come to the conclusion that it should either pad all or pad none. And as the complaint was that it added, I guess the only (already implemented) solution is to remove all spacing.

Reply:
Agreed, let's keep current behavior (removing spaces in all blank lines).

Reply:
Side note: except in those lines, where the indentation of the empty line is more than the heredoc indentation (marked by the heredoc end). For this case see lines 203-224.
These spaces can not be removed because this fixer is not marked as risky.